cmd/copy.go

```diff
@@ -876,13 +899,67 @@ func (cca *cookedCopyCmdArgs) processCopyJobPartOrders() (err error) {
 	// depending on the source and destination type, we process the cp command differently
 	// Create enumerator and do enumerating
 	switch cca.fromTo {
-	case common.EFromTo.LocalBlob():
-		fallthrough
-	case common.EFromTo.LocalBlobFS():
-		fallthrough
-	case common.EFromTo.LocalFile():
-		e := copyUploadEnumerator(jobPartOrder)
-		err = e.enumerate(cca)
+	case common.EFromTo.LocalBlob(),
+		common.EFromTo.LocalBlobFS(),
+		common.EFromTo.LocalFile():
+		var traverser resourceTraverser
+		var destTraverser resourceTraverser
+
+		dst := cca.destination
+
+		if cca.destinationSAS != "" {
+			destURL, err := url.Parse(dst)
+
+			if err != nil {
+				return err
+			}
+
+			destURL = copyHandlerUtil{}.appendQueryParamToUrl(destURL, cca.destinationSAS)
+			dst = destURL.String()
+		}
+
+		traverser, err = initResourceTraverser(cca.source, cca.fromTo.From(), nil, nil, &cca.followSymlinks, &cca.listOfFilesLocation, cca.recursive, func() {})
+		if err != nil {
+			return err
+		}
+
+		destTraverser, err = initResourceTraverser(dst, cca.fromTo.To(), &ctx, &cca.credentialInfo, &cca.followSymlinks, nil, cca.recursive, func() {})
```

Comment:
I initially felt confused why there was a destination traverser for a _copy_ operation. Maybe could consider make a routine called something like isDir(), which constructs this traverser, calls isDirectory on it (which is all we need) and throws it away.  To move that stuff out of this routine.

Reply:
I'm inclined to agree. I didn't necessarily _want_ to do this, but we also didn't have a unified way to tell if something was a directory or not. I'd tried to introduce one into copyHandlerUtil, but we all know that file is bloated to hell and back and doesn't need additional items.